history/taskana-simplehistory-provider/src/test/java/acceptance/events/task/CreateHistoryEventOnTaskRequestChangesAccTest.java

```diff
@@ -0,0 +1,81 @@
+package acceptance.events.task;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import acceptance.AbstractAccTest;
+import java.time.Instant;
+import java.util.List;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import pro.taskana.common.test.security.JaasExtension;
+import pro.taskana.common.test.security.WithAccessId;
+import pro.taskana.simplehistory.impl.SimpleHistoryServiceImpl;
+import pro.taskana.simplehistory.impl.TaskHistoryQueryImpl;
+import pro.taskana.simplehistory.impl.task.TaskHistoryQueryMapper;
+import pro.taskana.spi.history.api.events.task.TaskHistoryEvent;
+import pro.taskana.spi.history.api.events.task.TaskHistoryEventType;
+import pro.taskana.task.api.TaskService;
+import pro.taskana.task.api.TaskState;
+import pro.taskana.task.api.models.Task;
+
+@ExtendWith(JaasExtension.class)
+class CreateHistoryEventOnTaskRequestChangesAccTest extends AbstractAccTest {
+
+  private final TaskService taskService = taskanaEngine.getTaskService();
+  private final SimpleHistoryServiceImpl historyService = getHistoryService();
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_CreateClaimedHistoryEvent_When_TaskIsClaimed() throws Exception {
+    final String taskId = "TKI:100000000000000000000000000000000066";
+    final Instant oldModified = taskService.getTask(taskId).getModified();
+
+    TaskHistoryQueryMapper taskHistoryQueryMapper = getHistoryQueryMapper();
+
+    List<TaskHistoryEvent> events =
+        taskHistoryQueryMapper.queryHistoryEvents(
+            (TaskHistoryQueryImpl) historyService.createTaskHistoryQuery().taskIdIn(taskId));
+
+    assertThat(events).isEmpty();
+
+    assertThat(taskService.getTask(taskId).getState()).isEqualTo(TaskState.IN_REVIEW);
+    Task task = taskService.requestChanges(taskId);
+
+    assertThat(task.getState()).isEqualTo(TaskState.READY);
+
+    events =
+        taskHistoryQueryMapper.queryHistoryEvents(
+            (TaskHistoryQueryImpl) historyService.createTaskHistoryQuery().taskIdIn(taskId));
+
+    TaskHistoryEvent event = events.get(0);
+
+    assertThat(event.getEventType()).isEqualTo(TaskHistoryEventType.CHANGES_REQUESTED.getName());
+
+    event = historyService.getTaskHistoryEvent(event.getId());
+
+    assertThat(event.getDetails()).isNotNull();
+
+    JSONArray changes = new JSONObject(event.getDetails()).getJSONArray("changes");
+
+    JSONObject expectedModified =
+        new JSONObject()
+            .put("newValue", task.getModified().toString())
+            .put("fieldName", "modified")
+            .put("oldValue", oldModified.toString());
+    JSONObject expectedState =
+        new JSONObject()
+            .put("newValue", "READY")
+            .put("fieldName", "state")
+            .put("oldValue", "IN_REVIEW");
```

Comment:
suggestion: use TaskState.<value>.name() instead of writing the strings manually

Reply:
We do it this way in all tests that (as I remember). Would the inconsistency be ok?

Reply:
hmm. If we do it in all tests, then let's not do it here.
This can be fixed really fast using regex magic with IntelliJ. Maybe we can fix this everywhere later^^